config/clb-consumer/contact-data/delta-sync-dispatching.js

```diff
@@ -0,0 +1,123 @@
+        if(publicTypes.includes(type)) {
+          insertsOnPublic.push(`${insert.subject} ${insert.predicate} ${insert.object}.`)
+          //For the case where the subject is an admin unit but everything else must go in private graph
+          const contextGraphTriple = contextTriples.find((context) => context.predicate === '<http://mu.semte.ch/vocabularies/ext/contextDataGoesInGraph>' && context.object !== `<${LANDING_ZONE_GRAPH}>`)
+          for(let triple of otherContextTriples) {
+            if(contextGraphTriple) {
+              const graph = contextGraphTriple.object.slice(1,-1);
+              if(!insertsOnGraphs[graph]) {
+                insertsOnGraphs[graph] = [`${triple.subject} ${triple.predicate} ${triple.object}.`]
+              } else {
+                insertsOnGraphs[graph].push(`${triple.subject} ${triple.predicate} ${triple.object}.`)
+              }
+            } else {
+              insertsOnPublic.push(`${triple.subject} ${triple.predicate} ${triple.object}.`)
+            }
+          }
+        }
+      }
+    }
+    const deletesOnPublic = [];
+    const deletesOnGraphs = {};
+    for(let deletion of original.deletes) {
+      const subject = deletion.subject;
+      const contextTriples = withContext.deletes.filter((context) => context.subject === subject);
+      const graphTriple = contextTriples.find((context) => context.predicate === '<http://mu.semte.ch/vocabularies/ext/goesInGraph>' && context.object !== `<${LANDING_ZONE_GRAPH}>`)
+      const otherContextTriples = withContext.deletes.filter((context) => context.predicate !== '<http://mu.semte.ch/vocabularies/ext/goesInGraph>' && context.predicate !== '<http://mu.semte.ch/vocabularies/ext/contextDataGoesInGraph>' && !(context.subject === subject && context.predicate === '<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>'));
+      if(graphTriple) {
+        const graph = graphTriple.object.slice(1,-1); // We have to slice it to remove the "<" and ">"
+        if(!deletesOnGraphs[graph]) {
+          deletesOnGraphs[graph] = [`${deletion.subject} ${deletion.predicate} ${deletion.object}.`]
+        } else {
+          deletesOnGraphs[graph].push(`${deletion.subject} ${deletion.predicate} ${deletion.object}.`)
+        }
+        for(let triple of otherContextTriples) {
+          deletesOnGraphs[graph].push(`${triple.subject} ${triple.predicate} ${triple.object}.`)
+        }
+      }
+      const typeTriple = contextTriples.find((context) => context.subject === subject && context.predicate === '<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>');
+      if(!typeTriple) continue;
+      const type = typeTriple.object;
+      if(publicTypes.includes(type)) {
+        deletesOnPublic.push(`${deletion.subject} ${deletion.predicate} ${deletion.object}.`)
+        for(let triple of otherContextTriples) {
+          deletesOnPublic.push(`${triple.subject} ${triple.predicate} ${triple.object}.`)
+        }
+      }
+    }
```

Comment:
**Suggestion:** While the above seems functionally correct it is rather hard to parse/read imo. Would you mind adding some formatting such as blank lines and break up some long lines? Furthermore, several (if-then-else) blocks are nearly the same. Imo it could make the code easier to understand/maintain if we split these common parts into reusable functions.